### ISSUE TITLE
Add drop shadow support to Annotation Box

### ISSUE THREAD

**Issue by dov** [matplotlib/matplotlib#13720]
### Feature request

Add an option to for drawing a drop shadow below the AnnotationBox . Here are a a list of options that I would like to see for this feature:

- Color and alpha transparency of drop shadow
- x,y displacement of the drop shadow
- Gaussian bluring strength of the drop shadow (similar to blur% in inkscape)

**Comment by ImportanceOfBeingErnest** [comment]
Maybe this needs a bit more detail, concerning in how far [`matplotlib.patches.Shadow`](https://matplotlib.org/api/_as_gen/matplotlib.patches.Shadow.html) would not be sufficient.

**Comment by dov** [comment]
I fully agree it might just be my lack of understanding of how to use `Shadow` and would appreciate an example of how a `Shadow` of the same size as an `AnnotationBox` may be constructed.

**Comment by jklymak** [comment]
Untested, but `AnnotationBox` should have a `get_bbox_patch()` method, that then you can pass to `Shadow`.  A PR with a demo would be most welcome...

**Comment by ImportanceOfBeingErnest** [comment]
There is also [`matplotlib.patheffects.SimplePatchShadow`](https://matplotlib.org/api/patheffects_api.html#matplotlib.patheffects.withSimplePatchShadow). 
... which makes me wonder if both are really needed. 

In any case none of them allow for gaussian bluring - which is in general something for which matplotlib does not have any API.

**Comment by oscargus** [comment]
3.8 will add configuration of the pie (#25389) and legend (#24666) shadow. I guess this comes down to adding a `shadow` keyword argument to [FancyBboxPatch](https://matplotlib.org/stable/api/_as_gen/matplotlib.patches.FancyBboxPatch.html) in a similar manner? In that way one can pass `shadow={'ox': 3, ...}` as part of the `bbox` arguments (and this will also be applicable to quite a lot of other things relying on FancyBboxPatch).
